packages/core/src/browser/shell/application-shell.ts

```diff
@@ -873,6 +879,27 @@ export class ApplicationShell extends Widget {
         }
     }
 
+    /**
+     * Refresh the toggle button for the bottom panel. This implementation creates a status bar entry
+     * and refers to the command `core.toggle.bottom.panel`.
+     */
+    protected refreshBottomPanelToggleButton() {
+        if (this.bottomPanel.isEmpty) {
+            if (this.statusBar.hasElement(BOTTOM_PANEL_TOGGLE_ID)) {
```

Comment:
Why do you ask for `hasElement`? The statusbar was designed so you don't need to ask whether an element exists but simply set it.

Reply:
That leads to the status bar calling `update()` although nothing has changed. I wanted to avoid such unnecessary updates.

Reply:
It is an almost no-op since we use a virtual dom and no rerendering would happen if there was no change.

Reply:
Ok, I'll remove it.